### ISSUE TITLE
Adding a require during watch mode

### ISSUE THREAD

**Issue by jad-hamza** [epfl-lara/stainless#290]
In `--watch` mode, adding a require should regenerate the verification conditions.

To reproduce the issue:
- Copy the example in `File.scala` and run `stainless-scalac --watch File.scala`
- Uncomment the line with `require(false)`
- The verification condition for ̀`assert(false)` remains invalid, while it should become valid

```scala
object AddRequire {
  def lemma() = {
    // require(false)
    assert(false)
  }
}
```

**Comment by jad-hamza** [comment]
It's probably the same issue for the following example. When we uncomment the call `increment()`, the VC for the `assert(a == 0)`  stays valid while it should become invalid.

```scala
object WatchVar {
  case class A(var a: BigInt) {
    def increment() = {
      a = a + 1
    }

    def f() = {
      require(a == 0)

      // increment()
      assert(a == 0)
    }
  }
}
```

**Comment by jad-hamza** [comment]
There is also the same problem with adding/removing assertions, I'll look into it today

**Comment by jad-hamza** [comment]
From @samarion: the problem is that the pipeline caches are not invalidated (with the `invalidate` function)
One quick workaround (not tested) is to change the function [cached](https://github.com/epfl-lara/stainless/blob/b62000030079c226601e5ac03a2fec074ad489ef/core/src/main/scala/stainless/extraction/ExtractionPipeline.scala#L147) to just call builder.

**Comment by samarion** [comment]
I tried that at some point and it leads to non-termination in certain components

**Comment by samarion** [comment]
I tried again with the new caches and the bug still appeared, so this probably isn't a caching issue at the extraction pipeline level.

**Comment by samarion** [comment]
Fixed in 83d9469406b4a7284d46dea4ebb6dfb2a0aba571
